services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -85,10 +91,23 @@ class Scheduler(  # pylint: disable=too-many-instance-attributes, too-many-publi
     async def start(self) -> None:
         # run as a background task
         logger.info("Starting dynamic-sidecar scheduler")
-        self._keep_running = True
-        self._scheduler_task = asyncio.create_task(
-            self._run_scheduler_task(), name="dynamic-scheduler"
+
+        redis_clients_manager: RedisClientsManager = (
+            self.app.state.redis_clients_manager
         )
+
+        settings: DynamicServicesSchedulerSettings = (
+            self.app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SCHEDULER
+        )
+        self._scheduler_task = start_periodic_task(
+            exclusive(
+                redis_clients_manager.client(RedisDatabase.LOCKS),
+                lock_key="director-v2_dynamic-scheduler_task",
```

Comment:
This key needs to be unique, right? why not including `{__name__}.{self.__class__.__name__}` ? It is copy-past proof and also indicates where is its location